### PR TITLE
Add victim support for the projectile hit event

### PR DESCRIPTION
**Pull request by Mr-Darth**
### Description
Adds support for using `victim` in a projectile hit event

---
**Target Minecraft Versions:** 1.11+
**Requirements:** none
**Related Issues:** #3746

### CODE REVIEW

File: src/main/java/ch/njol/skript/expressions/ExprAttacked.java [hunk 1]
```diff
@@ -66,8 +73,8 @@
 
 	@Override
 	public boolean init(final Expression<?>[] vars, final int matchedPattern, final Kleenean isDelayed, final ParseResult parser) {
-		if (!ScriptLoader.isCurrentEvent(EntityDamageEvent.class, EntityDeathEvent.class, VehicleDamageEvent.class, VehicleDestroyEvent.class)) {
-			Skript.error("The expression 'victim' can only be used in a damage or death event", ErrorQuality.SEMANTIC_ERROR);
+		if (!ScriptLoader.isCurrentEvent(EntityDamageEvent.class, EntityDeathEvent.class, VehicleDamageEvent.class, VehicleDestroyEvent.class, ProjectileHitEvent.class)) {
+			Skript.error("The expression 'victim' can only be used in a damage, death, or projectile hit event", ErrorQuality.SEMANTIC_ERROR);
```
Comment (FranKusmiruk) [inline src/main/java/ch/njol/skript/expressions/ExprAttacked.java:77]: The error should take into account the version check as well

File: src/main/java/ch/njol/skript/expressions/ExprAttacked.java [hunk 2]
```diff
@@ -50,13 +52,18 @@
...
+@Description("The victim of a damage event, e.g. when a player attacks a zombie this expression represents the zombie. " +
+	"This also covers the hit entity in a projectile hit event.")
 @Examples({"on damage:",
 		"	victim is a creeper",
 		"	damage the attacked by 1 heart"})
-@Since("1.3")
-@Events({"damage", "death"})
+@Since("1.3, INSERT VERSION (projectile hit event)")
+@Events({"damage", "death", "projectile hit"})
+@RequiredPlugins("Minecraft 1.11+")
```
Comment (FranKusmiruk) [inline src/main/java/ch/njol/skript/expressions/ExprAttacked.java:62]: The @RequiredPlugins tag might be misleading on this expression considering it works on older versions, just not in the projectile hit event. You're probably better off just removing it and specifying the version requirement in the description.

**Review by FranKusmiruk — CHANGES_REQUESTED** [review]
other than what was commented, lgtm

File: src/main/java/ch/njol/skript/expressions/ExprAttacked.java [hunk 3]
```diff
@@ -90,7 +97,10 @@ public boolean init(final Expression<?>[] vars, final int matchedPattern, final
 		final Entity[] one = (Entity[]) Array.newInstance(type.getType(), 1);
 		Entity entity;
 		if (e instanceof EntityEvent)
-			entity = ((EntityEvent) e).getEntity();
+			if (e instanceof ProjectileHitEvent && SUPPORT_PROJECTILE_HIT)
```
Comment (Pikachu920) [inline src/main/java/ch/njol/skript/expressions/ExprAttacked.java:100]: minor nitpick but wouldn't it make more sense to check if project hit is supported before checking the class of `e`?

File: src/main/java/ch/njol/skript/expressions/ExprAttacked.java [hunk 4]
```diff
@@ -36,6 +37,7 @@
 import ch.njol.skript.doc.Events;
 import ch.njol.skript.doc.Examples;
 import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.RequiredPlugins;
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/expressions/ExprAttacked.java:40]: import can be removed. other than that looks good to me.

**Review by APickledWalrus — APPROVED** [review]

File: src/main/java/ch/njol/skript/expressions/ExprAttacked.java (hunk 4 above)
Comment (Mr-Darth) [reply src/main/java/ch/njol/skript/expressions/ExprAttacked.java]: Done! 👍

File: src/main/java/ch/njol/skript/expressions/ExprAttacked.java [hunk 5]
```diff
@@ -65,8 +70,8 @@
 
 	@Override
 	public boolean init(final Expression<?>[] vars, final int matchedPattern, final Kleenean isDelayed, final ParseResult parser) {
```
Comment (TPGamesNL) [inline src/main/java/ch/njol/skript/expressions/ExprAttacked.java:72]: Currently, if `SUPPORT_PROJECTILE_HIT` is false, `victim` is still usable in the projectile hit event, and it'll refer to the projectile itself

**Review by TPGamesNL — CHANGES_REQUESTED** [review]

**Review by TPGamesNL — APPROVED** [review]

**Pull request merged**